instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -10,36 +10,25 @@ module Redis
       module Patches
         # Module to prepend to Redis::Client for instrumentation
         module Client
-          def call(*args, &block)
-            response = nil
-
+          def process(commands)
+            reply = nil
             attributes = client_attributes
-            attributes['db.statement'] = Utils.format_statement(args)
+            attributes['db.statement'] = Utils.format_statements(commands)
             tracer.in_span(
-              Utils.format_command(args),
+              Utils.format_span_name(commands),
               attributes: attributes,
               kind: :client
-            ) do
-              response = super(*args, &block)
+            ) do |s|
+              reply = super(commands)
+              if reply.is_a?(::Redis::CommandError)
+                s&.record_exception(reply)
+                s&.status = Trace::Status.new(
```

Comment:
`s` should definitely be non-`nil` at this point.
```suggestion
                s.record_exception(reply)
                s.status = Trace::Status.new(
```